lib/iris/tests/unit/analysis/test_PERCENTILE.py

```diff
@@ -208,12 +208,25 @@ def test_masked(self):
         shape = (2, 11)
         data = ma.arange(np.prod(shape)).reshape(shape)
         data[0, ::2] = ma.masked
-        emsg = "Cannot use fast np.percentile method with masked array."
+        emsg = (
+            "Cannot use fast np.percentile method with masked array unless "
+            "mdtol is 0."
+        )
         with self.assertRaisesRegex(TypeError, emsg):
             PERCENTILE.aggregate(
                 data, axis=0, percent=50, fast_percentile_method=True
             )
 
+    def test_masked_mdtol_0(self):
+        shape = (2, 11)
+        axis = 0
+        percent = 50
+        data = ma.arange(np.prod(shape)).reshape(shape)
+        data[0, ::2] = ma.masked
+        expected = np.arange(shape[-1]) + 5.5
```

Comment:
```suggestion
        expected = ma.arange(shape[-1]) + 5.5
```

Reply:
Fixed (which also fixed the warnings I didn't understand about it ignoring this)